### PR TITLE
CORE-19749: Write "start requested" status to state manager

### PR DESCRIPTION
**Pull request by josephzunigadaly**
Write "start requested" to state manager so all REST workers recognise the flow ID when the start flow request completes.

### CODE REVIEW

File: components/flow/flow-rest-resource-service-impl/src/main/kotlin/net/corda/flow/rest/impl/FlowStatusLookupServiceImpl.kt [hunk 1]
```diff
@@ -99,6 +103,10 @@ class FlowStatusLookupServiceImpl @Activate constructor(
         }.start()
     }
 
+    override fun storeStatus(status: FlowStatus) {
```
Comment (LWogan) [inline components/flow/flow-rest-resource-service-impl/src/main/kotlin/net/corda/flow/rest/impl/FlowStatusLookupServiceImpl.kt:106]: Need a unit test

File: components/flow/flow-rest-resource-service-impl/src/main/kotlin/net/corda/flow/rest/impl/v1/FlowRestResourceImpl.kt [hunk 2]
```diff
@@ -205,9 +204,10 @@ class FlowRestResourceImpl @Activate constructor(
                         getKeyForStartEvent(status.key, holdingIdentityShortHash), startEvent
                     )
                 ),
-                Record(FLOW_STATUS_TOPIC, status.key, status),
             )
 
+            flowStatusLookupService.storeStatus(status)
```
Comment (LWogan) [inline components/flow/flow-rest-resource-service-impl/src/main/kotlin/net/corda/flow/rest/impl/v1/FlowRestResourceImpl.kt:209]: needs a unit test update

**Review by LWogan — COMMENTED** [review]
Looks fine to me. This addresses the problem.

**Comment by josephzunigadaly** [comment]
build e2e

File: components/flow/flow-rest-resource-service-impl/src/main/kotlin/net/corda/flow/rest/impl/FlowStatusLookupServiceImpl.kt (hunk 1 above)
Comment (josephzunigadaly) [reply components/flow/flow-rest-resource-service-impl/src/main/kotlin/net/corda/flow/rest/impl/FlowStatusLookupServiceImpl.kt:106]: Added tests - `FlowStatusLookupServiceImplTest.AfterInitialise.StateManagerWithContentAddedByStartFlow`

File: components/flow/flow-rest-resource-service-impl/src/main/kotlin/net/corda/flow/rest/impl/v1/FlowRestResourceImpl.kt (hunk 2 above)
Comment (josephzunigadaly) [reply components/flow/flow-rest-resource-service-impl/src/main/kotlin/net/corda/flow/rest/impl/v1/FlowRestResourceImpl.kt:209]: Added `storeStatus` call to mock verify in `FlowRestResourceImplTest`

**Comment by josephzunigadaly** [comment]
build e2e

**Comment by josephzunigadaly** [comment]
build e2e

**Review by LWogan — APPROVED** [review]
LGTM

**Pull request merged**